File: Data Structures & Algorithms/palindromic-substrings/submission-1.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.length();
        vector<vector<bool>> dp(n, vector<bool>(n, false));
        int ret = 0;
        int lIdx = 0, rIdx = n - 1;

        for (int i = n; i >= 0; i--) {
            for (int j = i; j < n; j++) {
                if (s[i] == s[j] & (j - i <= 2 || dp[i + 1][j - 1])) {
                    dp[i][j] = true;
                    ret++;
                }
            }
        }

        return ret;
    }
};

File: Data Structures & Algorithms/merge-intervals/submission-1.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ret = []
        intervals.sort(key = lambda x : x[0])

        curInterval = intervals[0]
        for i in range(1, len(intervals)):
            if intervals[i][0] > curInterval[1]:
                ret.append(curInterval)
                curInterval = intervals[i]
            elif intervals[i][1] < curInterval[0]:
                ret.append(intervals[i])
            else:
                curInterval = [min(intervals[i][0], curInterval[0]), max(intervals[i][1], curInterval[1])]
        
        ret.append(curInterval)
        return ret
        